lib/tests/streamlit/elements/image_test.py

```diff
@@ -123,39 +124,44 @@ def test_marshall_images(self, data_in, format):
         * Path
         * Bytes
         """
+        mimetype = f"image/{format}"
         file_id = _calculate_file_id(
             _np_array_to_bytes(data_in, output_format=format),
-            mimetype="image/" + format,
+            mimetype=mimetype,
         )
 
         st.image(data_in, output_format=format)
         imglist = self.get_delta_from_queue().new_element.imgs
         self.assertEqual(len(imglist.imgs), 1)
         self.assertTrue(imglist.imgs[0].url.startswith(MEDIA_ENDPOINT))
-        self.assertTrue(imglist.imgs[0].url.endswith(f".{format}"))
+        self.assertTrue(
+            imglist.imgs[0].url.endswith(get_extension_for_mimetype(mimetype))
+        )
         self.assertTrue(file_id in imglist.imgs[0].url)
 
     @parameterized.expand(
         [
-            (IMAGES["img_32_32_3_rgb"]["np"], "jpeg"),
-            (IMAGES["img_32_32_3_bgr"]["np"], "jpeg"),
-            (IMAGES["img_64_64_rgb"]["np"], "jpeg"),
-            (IMAGES["img_32_32_3_rgba"]["np"], "png"),
-            (IMAGES["img_32_32_3_rgb"]["pil"], "jpeg"),
-            (IMAGES["img_32_32_3_bgr"]["pil"], "jpeg"),
-            (IMAGES["img_64_64_rgb"]["pil"], "jpeg"),
-            (IMAGES["img_32_32_3_rgba"]["pil"], "png"),
+            (IMAGES["img_32_32_3_rgb"]["np"], ".jpg"),
+            (IMAGES["img_32_32_3_bgr"]["np"], ".jpg"),
+            (IMAGES["img_64_64_rgb"]["np"], ".jpg"),
+            (IMAGES["img_32_32_3_rgba"]["np"], ".png"),
+            (IMAGES["img_32_32_3_rgb"]["pil"], ".jpg"),
+            (IMAGES["img_32_32_3_bgr"]["pil"], ".jpg"),
+            (IMAGES["img_64_64_rgb"]["pil"], ".jpg"),
+            (IMAGES["img_32_32_3_rgba"]["pil"], ".png"),
         ]
     )
-    def test_marshall_images_with_auto_output_format(self, data_in, expected_format):
+    def test_marshall_images_with_auto_output_format(
+        self, data_in, expected_extension: str
```

Comment:
since you specified a type of `expected_extension` we can specify the type of `data_in` by `AtomicImage` from  (type defined in `image.py`) . But I don't mind not specifying types (especially complex types) in tests

Reply:
I can add that!